### PR TITLE
added kubernetes support

### PR DESCRIPTION
**Pull request by mschneider82**
This PR adds Kubernetes support for Kind Deployment to scale to zero.

### CODE REVIEW

**Comment by acouvreur** [comment]
Thanks for this PR, I think a test file would give me more confidence in the code be worked on later. If you could provide that, that'd be great.

**Comment by mschneider82** [comment]
I run into this issue https://github.com/kubernetes/kubernetes/issues/92195
the fake kubernetes mock doesnt support scale :-(

**Comment by mschneider82** [comment]
cannot get scaling working in tests. sorry, the fake client needs to support this otherwise it would be a large code hack to handle this

**Comment by mschneider82** [comment]
fixes #13

**Comment by acouvreur** [comment]
Can you by any change still add the tests and skip them with a reference to the issue you mentioned ?
It might get fixed later, would be great to have it.

**Review by acouvreur — CHANGES_REQUESTED** [review]

File: pkg/scaler/docker_kubernetes.go [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+		if err != nil {
+			log.Error(err.Error())
+			return err
+		}
+
+		sc := *s
+		if sc.Spec.Replicas == 0 {
+			sc.Spec.Replicas = int32(config.Replicas)
+		} else {
+			log.Infof("Replicas for %s %s in namespace %s are already: %d", config.Kind, config.Name, config.Namespace, sc.Spec.Replicas)
```
[the remark this answers is not in the document]
Comment (mschneider82) [reply pkg/scaler/docker_kubernetes.go:77]: this happens quite often, so i think this is a good thing

File: main.go [hunk 2]
```diff
@@ -23,29 +25,50 @@ type OnDemandRequestState struct {
 func main() {
 
 	swarmMode := flag.Bool("swarmMode", true, "Enable swarm mode")
+	kubernetesMode := flag.Bool("kubernetesMode", false, "Enable Kubernetes mode")
 
 	flag.Parse()
 
-	dockerScaler := getDockerScaler(*swarmMode)
+	dockerScaler := getDockerScaler(*swarmMode, *kubernetesMode)
```
[the remark this answers is not in the document]
Comment (mschneider82) [reply main.go:32]: well, i dont think that we should take care of this configuration mistake. But that swarmmode is defaulted to `true` should be changed... ;-)

[the remark this answers is not in the document]
Comment (acouvreur) [reply main.go:32]: Hah, I will probably change it later as a breaking change

**Review by acouvreur — APPROVED** [review]

**Pull request merged**